cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/common/DataprocMetric.java

```diff
@@ -29,19 +29,37 @@ public class DataprocMetric {
   private final Exception exception;
   @Nullable
   private final LaunchMode launchMode;
+  @Nullable
+  private final String imageVersion;
 
   private DataprocMetric(String metricName, String region, @Nullable Exception exception,
-      @Nullable LaunchMode launchMode) {
+      @Nullable LaunchMode launchMode, @Nullable String imageVersion) {
     this.metricName = metricName;
     this.region = region;
     this.exception = exception;
     this.launchMode = launchMode;
+    this.imageVersion = imageVersion;
   }
 
   public String getMetricName() {
     return metricName;
   }
 
+  @Nullable
+  public String getImageVersion() {
+    if (imageVersion != null) {
+      // return major.minor
+      int firstDotIdx = imageVersion.indexOf('.');
```

Comment:
May be splitting and combining will be more simple and readable

Reply:
ack, done.